### PR TITLE
Integrated CRUD with front end for various components 

### PR DESCRIPTION
**Pull request by alexander-huerta**
[x] GET profile integrated with the front end. Currently hardcoded to retrieve a certain ID. Next step is to use the ID of the currently logged in user. 
get_profile

[x] Small typo fixed in PostRide component 

[x] GET rides integrated. The param accepted  is currently the origin city and is case sensitive. 
get_rides

[x] GET reviews added into the front end code. It is currently returning an array of reviews. Next step is for the component owner to handle the last step of rendering this received data instead of the local dummy data. Next step is to use the ID of the target user. 

[x] POST reviews integrated and properly saving new reviews to the database. The next step, on a similar note is to add the ID of the reviewer dynamically. The input id is currently hard coded. 
get_profile

### CODE REVIEW

File: client/src/components/Messages.jsx [hunk 1]
```diff
@@ -12,10 +12,22 @@ import Message from './Message.jsx';
...
 import dummyData from '../../../messagesDummyData';
 
+import { AuthContext } from '../contexts/index.jsx';
+
+
 function Messages({ userId }) {
   const [usersContacted, setUsersContacted] = useState(dummyData.usersContacted);
   const [messages, setMessages] = useState(dummyData.messages);
   const [userContactedId, setUserContactedId] = useState(null);
+  //  alex test
```
Comment (aaronfinsrud) [inline client/src/components/Messages.jsx:22]: Any chance we can clean these comments up before merging?

File: client/src/contexts/index.jsx [hunk 2]
```diff
@@ -59,21 +67,35 @@ export function AuthProvider({ children }) {
...
+      reviews.push(review.data());
+    });
+    return reviews;
   }
 
   function addMessage(body) {
     return addDoc(messagesCollectionReference, body);
   }
+  // function getMessages(params) {
+  //   return getDocs(messagesCollectionReference, params);
```
Comment (aaronfinsrud) [inline client/src/contexts/index.jsx:97]: I think you might still be working on this.

**Review by aaronfinsrud — APPROVED** [review]
Other than a few lingering comments -- it looks good to me. Are you keeping track of the routes that you've implemented and the possible improvements for each? Cheers

File: client/src/contexts/index.jsx (hunk 2 above)
Comment (alexander-huerta) [reply client/src/contexts/index.jsx:97]: deleting now!

File: client/src/components/Messages.jsx (hunk 1 above)
Comment (alexander-huerta) [reply client/src/components/Messages.jsx:22]: removing the comments now, I meant to remove before this PR

**Comment by alexander-huerta** [comment]
Yes, im keeping track. I can share the checklist tomorrow

**Pull request merged**